relayer/processor/types.go

```diff
@@ -68,75 +94,67 @@ func (channelKey ChannelKey) Counterparty() ChannelKey {
 }
 
-// ChannelStateCache maintains channel state for multiple channels.
-type ChannelStateCache map[ChannelKey]ChannelState
-
-// Flush is used to make a copy of the ChannelStateCache, helpful for the ChainProcessors
-// to clone their local cache so that PathProcessors can have a thread-safe copy.
-// It will also empty the messages for the existing channel state.
-func (c ChannelStateCache) Flush() ChannelStateCache {
-	copy := make(ChannelStateCache)
-	for channelKey, existingState := range c {
-		copy[channelKey] = existingState
-		flushed := existingState
-		flushed.Messages = nil
-		c[channelKey] = flushed
+// Counterparty flips a ConnectionKey for the perspective of the counterparty chain
+func (connectionKey ConnectionKey) Counterparty() ConnectionKey {
+	return ConnectionKey{
+		ClientID:                 connectionKey.CounterpartyClientID,
+		ConnectionID:             connectionKey.CounterpartyConnectionID,
+		CounterpartyClientID:     connectionKey.ClientID,
+		CounterpartyConnectionID: connectionKey.ConnectionID,
 	}
-	return copy
 }
 
+// ChannelStateCache maintains channel open state for multiple channels.
+type ChannelStateCache map[ChannelKey]bool
+
 // Merge will merge another ChannelStateCache into this one, appending messages and updating the Open state.
 func (c ChannelStateCache) Merge(other ChannelStateCache) {
 	for channelKey, newState := range other {
-		existingState, ok := c[channelKey]
-		if !ok {
-			c[channelKey] = newState
-			return
-		}
-		existingState.Messages = append(existingState.Messages, newState.Messages...)
-		existingState.Open = newState.Open
-		c[channelKey] = existingState
+		c[channelKey] = newState
+	}
+}
+
+// ConnectionStateCache maintains connection open state for multiple connections.
+type ConnectionStateCache map[ConnectionKey]bool
+
+// Merge will merge another ChannelStateCache into this one, appending messages and updating the Open state.
+func (c ConnectionStateCache) Merge(other ConnectionStateCache) {
+	for channelKey, newState := range other {
+		c[channelKey] = newState
 	}
 }
 
 // ChainProcessorCacheData is the data sent from the ChainProcessors to the PathProcessors
 // to keep the PathProcessors up to date with the latest info from the chains.
 type ChainProcessorCacheData struct {
-	ChannelMessageCache
+	IBCMessagesCache
 	InSync bool
+	ConnectionStateCache
 	ChannelStateCache
 }
 
-// Merge will merge another MessageCache into this one.
-func (c MessageCache) Merge(other MessageCache) {
-	for k, v := range other {
-		c[k] = v
-	}
-}
-
-// Clone will create a deep copy of a MessageCache.
-func (c MessageCache) Clone() MessageCache {
-	newMessageCache := make(MessageCache)
+// Clone will create a deep copy of a PacketMessagesCache.
+func (c PacketMessagesCache) Clone() PacketMessagesCache {
+	newPacketMessagesCache := make(PacketMessagesCache)
 	for mk, mv := range c {
-		newSequenceCache := make(SequenceCache)
+		newPacketSequenceCache := make(PacketSequenceCache)
```

Comment:
Slightly better if these `make` calls can use size hints to avoid regrowing the maps.

Reply:
done